### PR TITLE
python-freethreading: update livecheck and sync with py3.13 formulae

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/p/python-freethreading.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class PythonFreethreading < Formula
 
   livecheck do
     url "https://www.python.org/ftp/python/"
```
Comment (samford) [inline Formula/p/python-freethreading.rb:9]:
```suggestion
    formula "python"
```

File: Formula/p/python-freethreading.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ class PythonFreethreading < Formula
...
-    regex(%r{href=.*?v?(\d(?:\.\d+)*)/?["' >]}i)
+    regex(%r{href=.*?v?(3\.13(?:\.\d+)*)/?["' >]}i)
```
Comment (samford) [inline Formula/p/python-freethreading.rb:10]:
```suggestion
```

**Review by samford — COMMENTED** [review]
This isn't a versioned formula, so we may get into a situation where the version that's used in the regex isn't kept in sync with the latest stable version of Python in this case. Unless I'm missing something (e.g., technical restrictions, etc.), referencing the `python` alias would achieve the same goal without requiring anyone to remember to manually update this regex when we package a newer Python version.

**Comment by chenrui333** [comment]
good call, using python is a better choice

**Review by samford — APPROVED** [review]
Assuming we can't do `["python", "python-freethreading"]` in `synced_versions_formulae.json`, we'll just have to remember to move `python-freethreading` into the newest Python array when we create formulae for a newer Python version.

**Comment by chenrui333** [comment]
> we'll just have to remember to move `python-freethreading` into the newest Python array when we create formulae for a newer Python version.

yeah, I can imagine that would be popped up for py3.14

**Pull request merged**